### PR TITLE
Update network SDK to track2 

### PR DESCRIPTION
**Pull request by bitoku**
### Which issue this PR addresses:

Fixes https://issues.redhat.com/browse/ARO-4665

### What this PR does / why we need it:

This PR introduces new track2 SDK in cluster manager.
It relates to https://github.com/Azure/ARO-RP/pull/3568

This is blocked by https://github.com/Azure/ARO-RP/pull/3579

### Test plan for issue:

This doesn't change/add any new behaviours.
Just compile check and code review is enough.

### Is there any documentation that needs to be updated for this PR?

tech debt cleanup, N/A

### How do you know this will function as expected in production? 

This PR itself doesn't change/add any new behaviours.

### CODE REVIEW

File: pkg/util/azureclient/azuresdk/common/options.go [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+	"github.com/Azure/go-autorest/autorest"
+)
+
+const (
+	ErrCodeInvalidClientSecretProvided = "AADSTS7000215" // https://login.microsoftonline.com/error?code=7000215
+	ErrCodeMissingRequiredParameters   = "AADSTS7000216" // https://login.microsoftonline.com/error?code=7000216
+	AuthorizationFailed                = "AuthorizationFailed"
+)
```
Comment (AldoFusterTurpin) [inline pkg/util/azureclient/azuresdk/common/options.go:19]: Thank you for applying the suggestion, this really helps 🙂

**Review by AldoFusterTurpin — APPROVED** [review]
Even it is in Draft as it depends on your other PR, LGTM. Thank you for this!

**Review by AldoFusterTurpin — APPROVED** [review]
LGTM! Thanks.

**Comment by bitoku** [comment]
There seems something wrong with tenant ID.
I made it draft for the time being.

**Review by mociarain — APPROVED** [review]
LGTM

**Review by AldoFusterTurpin — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**